File: sdk/samples/WCFSamples/TechnologySamples/Scenario/interop/UDDI/VB/UddiClient.vb
' Copyright (C) 2003-2005 Microsoft Corporation, All rights reserved.

Imports System
Imports System.IO
Imports System.Text
Imports System.ServiceModel
Imports Microsoft.VisualBasic

Namespace Microsoft.ServiceModel.Samples

    Public Class UddiClient

        Private inquireProxy As InquireClient
        Private publishProxy As PublishClient
        Private token As authToken

        Public Sub New()

            inquireProxy = New InquireClient("Inquire")
            publishProxy = New PublishClient("Publish")

        End Sub

        Public Sub Login(ByVal username As String, ByVal password As String)

            Dim getToken As New get_authToken()
            getToken.generic = "2.0"
            getToken.userID = username
            getToken.cred = password

            Dim request As New get_authTokenRequest()
            request.get_authToken = getToken
            token = publishProxy.get_authToken(request).authToken

        End Sub

        Public Function GetBusinessByName(ByVal businessName As String) As businessInfo()

            Dim find As New find_business()
            find.name = New name(0) {New name()}
            find.name(0).Value = businessName
            find.generic = "2.0"
            find.maxRows = 20

            Dim request As New find_businessRequest()
            request.find_business = find
            Dim bList As businessList = inquireProxy.find_business(request).businessList
            Return bList.businessInfos

        End Function

        Public Function PublishBusiness(ByVal businessName As String, ByVal description As String) As String

            Dim newBusiness As New save_business()
            newBusiness.authInfo = token.authInfo
            newBusiness.generic = "2.0"
            newBusiness.businessEntity = New businessEntity(0) {New businessEntity()}

            Dim myBusinessEntity As businessEntity = newBusiness.businessEntity(0)
            myBusinessEntity.name = New name(0) {New name()}
            myBusinessEntity.name(0).Value = businessName
            myBusinessEntity.businessKey = ""
            myBusinessEntity.description = New description(0) {New description()}
            myBusinessEntity.description(0).Value = description

            Dim request As New save_businessRequest()
            request.save_business = newBusiness
            Dim myBusinessDetail As businessDetail = publishProxy.save_business(request).businessDetail

            Return myBusinessDetail.businessEntity(0).businessKey

        End Function

        Public Function GetMyBusinesses() As businessInfo()

            Dim getInfo As New get_registeredInfo()

            getInfo.generic = "2.0"
            getInfo.authInfo = token.authInfo

            Dim request As New get_registeredInfoRequest()
            request.get_registeredInfo = getInfo
            Dim info As registeredInfo = publishProxy.get_registeredInfo(request).registeredInfo
            If info Is Nothing Then
                Return Nothing
            End If

            Return info.businessInfos

        End Function

        Public Function GetMyBusinessByName(ByVal businessName As String) As businessInfo

            Dim getInfo As New get_registeredInfo()

            getInfo.generic = "2.0"
            getInfo.authInfo = token.authInfo

            Dim request As New get_registeredInfoRequest()
            request.get_registeredInfo = getInfo
            Dim info As registeredInfo = publishProxy.get_registeredInfo(request).registeredInfo

            If info Is Nothing Then
                Return Nothing
            End If

            For Each bInfo As businessInfo In info.businessInfos

                If bInfo.name(0).Value = businessName Then

                    Return bInfo

                End If

            Next

            Return Nothing

        End Function

        Public Sub DeleteMyBusinesses()

            Dim infos() As businessInfo = Me.GetMyBusinesses()

            Dim business As New delete_business()
            business.businessKey = New String(infos.Length - 1) {}
            For i As Integer = 0 To infos.Length - 1

                business.businessKey(i) = infos(i).businessKey

            Next
            business.generic = "2.0"
            business.authInfo = token.authInfo

            Dim request As New delete_businessRequest()
            request.delete_business = business
            publishProxy.delete_business(request)

        End Sub

        Public Sub DeleteMyBusiness(ByVal businessKey As String)

            Dim business As New delete_business()
            business.businessKey = New String(0) {businessKey}
            business.generic = "2.0"
            business.authInfo = token.authInfo

            Dim request As New delete_businessRequest()
            request.delete_business = business
            publishProxy.delete_business(request)

        End Sub

        Public Sub PrintBusinessInfo(ByVal bInfo As businessInfo)

            If bInfo.name IsNot Nothing Then

                For Each n As name In bInfo.name
                    Console.WriteLine("Name:        " & n.Value)
                Next

            End If

            If bInfo.description IsNot Nothing Then

                For Each desc As description In bInfo.description
                    Console.WriteLine("Description: " & desc.Value)
                Next

            End If
            Console.WriteLine("Key:         " & bInfo.businessKey)

            If bInfo.serviceInfos IsNot Nothing Then
                PrintServiceInfos(bInfo.serviceInfos)
            End If

            Console.WriteLine()

        End Sub

        Public Sub PrintServiceInfos(ByVal serviceInfos As serviceInfo())

            Console.WriteLine("Total Services Offered = " & serviceInfos.Length.ToString())
            If serviceInfos.Length > 0 Then

                For Each sInfo As serviceInfo In serviceInfos

                    Console.WriteLine(vbTab & "Service Key: " & sInfo.serviceKey)

                    If sInfo.name IsNot Nothing Then

                        For Each n As name In sInfo.name
                            Console.WriteLine(vbTab & "Name: " & n.Value)
                        Next

                    End If

                    Console.WriteLine()

                Next

            End If

        End Sub

        Public Sub Close()

            DirectCast(inquireProxy, ICommunicationObject).Close()
            DirectCast(publishProxy, ICommunicationObject).Close()

        End Sub

    End Class

End Namespace

File: sdk/samples/WCFSamples/TechnologySamples/Scenario/interop/UDDI/CS/UddiClient.cs
#region Copyright Notice
// ----------------------------------------------------------------------------
// Copyright (C) 2003-2005 Microsoft Corporation, All rights reserved.
// ----------------------------------------------------------------------------

#endregion

using System;
using System.IO;
using System.Text;
using System.ServiceModel;

namespace Microsoft.ServiceModel.Samples
{
    

    public class UddiClient 
    {
        InquireClient inquireProxy;
        PublishClient publishProxy;
        authToken token;        

        public UddiClient()
        {
            inquireProxy = new InquireClient("Inquire");
            publishProxy = new PublishClient("Publish");
        }

        public void Login(string username, string password)
        {
            get_authToken getToken = new get_authToken();
            getToken.generic = "2.0";
            getToken.userID = username;
            getToken.cred = password;

            
            token = publishProxy.get_authToken(getToken);
        }

        public businessInfo[] GetBusinessByName(string businessName)
        {
            find_business find = new find_business();
            find.name = new name[1] { new name() };
            find.name[0].Value = businessName;
            find.generic = "2.0";
            find.maxRows = 20;
                        
            businessList bList = inquireProxy.find_business(find);
            return bList.businessInfos;
        }


        public string PublishBusiness(string businessName, string description)
        {
            save_business newBusiness = new save_business();
            newBusiness.authInfo = token.authInfo;
            newBusiness.generic = "2.0";
            newBusiness.businessEntity = new businessEntity[1] { new businessEntity() };

            businessEntity myBusinessEntity = newBusiness.businessEntity[0];
            myBusinessEntity.name = new name[1] { new name() };
            myBusinessEntity.name[0].Value = businessName;
            myBusinessEntity.businessKey = "";
            myBusinessEntity.description = new description[1] { new description() };
            myBusinessEntity.description[0].Value = description;
                        
            businessDetail myBusinessDetail = publishProxy.save_business(newBusiness);

            return myBusinessDetail.businessEntity[0].businessKey;
        }

        public businessInfo[] GetMyBusinesses()
        {
            get_registeredInfo getInfo = new get_registeredInfo();

            getInfo.generic = "2.0";
            getInfo.authInfo = token.authInfo;
            
            registeredInfo info = publishProxy.get_registeredInfo(getInfo);
            if (info == null) return null;

            return info.businessInfos;
        }

        public businessInfo GetMyBusinessByName(string businessName)
        {
            businessInfo[] myBusinesses = this.GetMyBusinesses();

            foreach (businessInfo bInfo in myBusinesses)
            {
                if (bInfo.name[0].Value == businessName)
                {
                    return bInfo;
                }
            }
            return null;
        }

        public void DeleteMyBusinesses()
        {
            businessInfo[] myBusinesses = this.GetMyBusinesses();

            delete_business business = new delete_business();
            business.businessKey = new string[myBusinesses.Length];
            for (int i = 0; i < myBusinesses.Length; i++)
            {
                business.businessKey[i] = myBusinesses[i].businessKey;
            }
            business.generic = "2.0";
            business.authInfo = token.authInfo;
                        
            publishProxy.delete_business(business);
        }

        public void DeleteMyBusiness(string businessKey)
        {
            delete_business business = new delete_business();
            business.businessKey = new string[1] { businessKey };
            business.generic = "2.0";
            business.authInfo = token.authInfo;
                        
            publishProxy.delete_business(business);
        }

        public void PrintBusinessInfo(businessInfo bInfo)
        {
            if (bInfo.name != null)
            {
                foreach (name n in bInfo.name)
                    Console.WriteLine("Name:        " + n.Value);
            }

            if (bInfo.description != null)
            {
                foreach (description desc in bInfo.description)
                    Console.WriteLine("Description: " + desc.Value);
            }
            Console.WriteLine("Key:         " + bInfo.businessKey);

            if (bInfo.serviceInfos != null)
                PrintServiceInfos(bInfo.serviceInfos);

            Console.WriteLine();
        }

        public void PrintServiceInfos(serviceInfo[] serviceInfos)
        {
            Console.WriteLine("Total Services Offered = " + serviceInfos.Length);
            if (serviceInfos.Length > 0)
            {
                foreach (serviceInfo sInfo in serviceInfos)
                {
                    Console.WriteLine("\tService Key: " + sInfo.serviceKey);

                    if (sInfo.name != null)
                    {
                        foreach (name n in sInfo.name)
                            Console.WriteLine("\tName: " + n.Value);

                    }

                    Console.WriteLine();
                }
            }
        }
        
        public void Close()
        {
            ((ICommunicationObject)inquireProxy).Close();
            ((ICommunicationObject)publishProxy).Close();
        }

    }

    
}
